pkg/build/generator/generator.go

```diff
@@ -210,6 +259,19 @@ func (g *BuildGenerator) resolveImageStreamReference(ctx kapi.Context, from *kap
 
 	glog.V(4).Infof("Resolving ImageStreamReference %s of Kind %s in namespace %s", from.Name, from.Kind, namespace)
 	switch from.Kind {
+	case "ImageStream":
+		// NOTE: The 'ImageStream' reference should be used only for the 'output' image
```

Comment:
@soltysh to handle Output.To as that might be ImageStream. We should refactor this at some point and have some unified helper method inside pkg/image. Feel free to create follow up issue for that.